blocks/form/gdpr-confirmation.js

```diff
@@ -5,31 +5,33 @@ function hasQueryParams() {
 }
 
 // Send a query parameters to the GDPR API Accept
-async function acceptResult(accept, urlProd, sucessRedirect, block) {
-  // get url parameters
-  const urlParams = new URLSearchParams(window.location.search);
-  const paramData = urlParams.get('data');
-  const paramInit = urlParams.get('init');
-  const url = urlProd + accept;
-  const options = {
-    method: 'PUT',
-    headers: { 'Content-type': 'application/json; charset=UTF-8' },
-    body: JSON.stringify({
-      data: paramData,
-      init: paramInit,
-      confirm: 'true',
-    }),
-  };
+async function acceptResult(accept, urlProd, sucessRedirect, block, submitButton) {
   try {
+    submitButton.disabled = true;
+    // get url parameters
+    const urlParams = new URLSearchParams(window.location.search);
+    const paramData = urlParams.get('data');
+    const paramInit = urlParams.get('init');
+    const url = urlProd + accept;
+    const options = {
+      method: 'PUT',
+      headers: { 'Content-type': 'application/json; charset=UTF-8' },
+      body: JSON.stringify({
+        data: paramData,
+        init: paramInit,
+        confirm: 'true',
+      }),
+    };
     const response = await fetch(url, options);
     const resData = await response.text();
     if (response.ok) {
-      window.open(sucessRedirect);
+      window.location.href = sucessRedirect;
     } else {
       const { message } = JSON.parse(resData);
       throw message;
     }
   } catch (error) {
+    [...block.querySelectorAll('.error')].forEach((div) => div.remove());
```

Comment:
querySelectorAll already returns a list that have forEach available. no need to destructure it